### PR TITLE
Update lib-external-provisioner and cleanup constraints

### PR DESCRIPTION
**Pull request by msau42**
This is needed for https://github.com/kubernetes-csi/external-provisioner/pull/238

### CODE REVIEW

**Comment by msau42** [comment]
I was able to get past conflicts with lib-external-provisioner, but now I'm hitting conflicts with external-snapshotter.

@jsafrane can you help cleanup the constraints in external-snapshotter?

File: Gopkg.toml [hunk 1]
```diff
@@ -24,41 +24,37 @@
...
-[[constraint]]
-  name = "github.com/kubernetes-csi/external-snapshotter"
-  version = "1.0.0-rc4"
+#[[constraint]]
+#  name = "github.com/kubernetes-csi/external-snapshotter"
+#  version = "1.0.0-rc4"
 
-[[constraint]]
+[[override]]
   name = "github.com/kubernetes-sigs/sig-storage-lib-external-provisioner"
```
Comment (ddebroy) [inline Gopkg.toml]: This `name` may also need to switch to `sigs.k8s.io/sig-storage-lib-external-provisioner`. I kept running `dep ensure` failures without the above switch - so just wanted to call it out.

**Comment by jsafrane** [comment]
```
vendor/sigs.k8s.io/sig-storage-lib-external-provisioner/controller/controller.go:805:30: too many arguments in call to resourcelock.New
	have (string, string, string, "github.com/kubernetes-csi/external-provisioner/vendor/k8s.io/client-go/kubernetes/typed/core/v1".CoreV1Interface, nil, resourcelock.ResourceLockConfig)
	want (string, string, string, "github.com/kubernetes-csi/external-provisioner/vendor/k8s.io/client-go/kubernetes/typed/core/v1".CoreV1Interface, resourcelock.ResourceLockConfig)
```

This comes from lib-external-provisioner using master client-go. Update to v1.14.0-beta.1 should help, however, you need to update external-snapshotter first.

**Comment by msau42** [comment]
Update: Things compile now.  I'm trying to run the hostpath e2es with my private image and many are failing.  Am debugging...

**Comment by msau42** [comment]
```
39493 csi-hostpath-provisioner-0/csi-provisioner: /csi-provisioner flag redefined: log_dir
39494 csi-hostpath-provisioner-0/csi-provisioner: panic: /csi-provisioner flag redefined: log_dir
39495 csi-hostpath-provisioner-0/csi-provisioner: 
39496 csi-hostpath-provisioner-0/csi-provisioner: goroutine 1 [running]:
39497 csi-hostpath-provisioner-0/csi-provisioner: flag.(*FlagSet).Var(0xc0000a0120, 0x1688760, 0x23b3a10, 0x14e2f18, 0x7, 0x15139f8, 0x2f)
39498 csi-hostpath-provisioner-0/csi-provisioner:     /usr/local/google/home/msau/.gvm/gos/go1.11.1/src/flag/flag.go:805 +0x529
39499 csi-hostpath-provisioner-0/csi-provisioner: flag.(*FlagSet).StringVar(0xc0000a0120, 0x23b3a10, 0x14e2f18, 0x7, 0x0, 0x0, 0x15139f8, 0x2f)
39500 csi-hostpath-provisioner-0/csi-provisioner:     /usr/local/google/home/msau/.gvm/gos/go1.11.1/src/flag/flag.go:708 +0x8a
39501 csi-hostpath-provisioner-0/csi-provisioner: github.com/kubernetes-csi/external-provisioner/vendor/k8s.io/klog.InitFlags(0xc0000a0120)
39502 csi-hostpath-provisioner-0/csi-provisioner:     /usr/local/google/home/msau/workspace/go/src/github.com/kubernetes-csi/external-provisioner/vendor/k8s.io/klog/klog.go:411 +0x7b
39503 csi-hostpath-provisioner-0/csi-provisioner: main.init.0()
39504 csi-hostpath-provisioner-0/csi-provisioner:     /usr/local/google/home/msau/workspace/go/src/github.com/kubernetes-csi/external-provisioner/cmd/csi-provisioner/csi-provisioner.go:75 +0x13e
```

**Comment by msau42** [comment]
This may be because glog got pulled in again for some reason...

**Comment by msau42** [comment]
actually no, glog was already in vendor before this change

**Comment by msau42** [comment]
Trying again with just "dep ensure" instead of "dep ensure -update".

**Comment by msau42** [comment]
Tested and passed with PD driver.

I'm still having issues with hostpath e2es but I think it's unrelated to this change.

**Comment by pohly** [comment]
Michelle Au <notifications@github.com> writes:
> 39493 csi-hostpath-provisioner-0/csi-provisioner: /csi-provisioner flag redefined: log_dir
> 39494 csi-hostpath-provisioner-0/csi-provisioner: panic:
> /csi-provisioner flag redefined: log_dir

I've seen such issues when `vendor` wasn't complete and something was
pulled in from my GOPATH.

File: Gopkg.toml [hunk 2]
```diff
@@ -20,46 +20,32 @@
...
+# TODO: can this be removed?
 [[constraint]]
   name = "google.golang.org/grpc"
   version = "1.9.2"
 
-[[constraint]]
-  name = "k8s.io/apimachinery"
-  version = "kubernetes-1.13.0-beta.1"
-
+# TODO: remove when official 11.0.0 tagged
```
Comment (jsafrane) [inline Gopkg.toml]: official *1.14.0* ?

Comment (msau42) [reply Gopkg.toml]: Client-go has a separate versioning scheme than kubernetes. 11.0 corresponds to 1.14

Comment (jsafrane) [reply Gopkg.toml]: ack

**Pull request merged**